core/arch/arm/pta/benchmark.c

```diff
@@ -43,6 +43,21 @@
 
 struct tee_ts_global *bench_ts_global;
 static struct mutex bench_reg_mu = MUTEX_INITIALIZER;
+static uint32_t prev_cntkctl;
+
+static void timer_enable_el0_access(void)
+{
+	uint32_t cntkctl;
+
+	prev_cntkctl = read_cntkctl();
```

Comment:
By registering twice, `prev_cntkctl` will be overwritten and the access will be permanently enabled.

Reply:
@jenswi-linaro, right, `timer_enable_el0_access();` call should be after this check in `rpc_reg_global_buf`. thanks!
```
/* Check if we have already registered buffer */
	if (bench_ts_global) {
		EMSG("Timestamp buffer was already registered\n");
		mutex_unlock(&bench_reg_mu);
		return TEE_ERROR_BAD_STATE;
	}
```

Reply:
@jenswi-linaro fixed